OPNsense/opnsense/ingest/parser.yml

```
@@ -0,0 +1,60 @@
+name: 'opnsense-package-filter'
+pipeline:
+  - name: parsed_event
+    external:
+      name: grok.match
+      properties:
+        input_field: '{{original.message}}'
+        output_field: message
+        pattern: '%{PACKET_FILTER},(%{PF_IPV4}|%{PF_IPV6}),(%{PF_UDP}|%{PF_TCP}|%{PF_CARP}|%{PF_ICMP})'
+
+        custom_patterns:
+          PACKET_FILTER: '%{NUMBER:rulenr},%{DATA:subrulenr},%{DATA:anchorname},%{WORD:label},%{WORD:interface},%{WORD:reason},%{WORD:action},%{WORD:dir}'
+          PF_IPV4: '4,%{WORD:tos},%{DATA:ecn},%{NUMBER:ttl},%{NUMBER:id},%{NUMBER:offset},%{WORD:flags},%{NUMBER:protonum},%{WORD:protoname},%{NUMBER:length},%{IPV4:src},%{IPV4:dst}'
+          PF_IPV6: '6,%{DATA:class},%{DATA:flow},%{DATA:hoplimit},%{WORD:protoname},%{NUMBER:protonum},{NUMBER:length},%{IPV6:src},%{IPV6:dst}'
+          PF_UDP: '%{NUMBER:srcport},%{NUMBER:dstport},%{NUMBER:datalen}'
+          PF_TCP: '%{NUMBER:srcport},%{NUMBER:dstport},%{NUMBER:datalen},%{WORD:flags},%{NUMBER:seq},%{DATA:ack},%{NUMBER:window},%{DATA:urg},%{WORD:options}'
```

I suggest renaming `flags` to `tcp_flags` to avoid collision with `flags` extracted from the IPv4 pattern
```suggestion
          PF_TCP: '%{NUMBER:srcport},%{NUMBER:dstport},%{NUMBER:datalen},%{WORD:tcp_flags},%{NUMBER:seq},%{DATA:ack},%{NUMBER:window},%{DATA:urg},%{WORD:options}'
```